packages/clerk-js/src/ui/elements/FormControl.tsx

```diff
@@ -387,10 +387,24 @@ export const FormControl = forwardRef<HTMLInputElement, PropsWithChildren<FormCo
     <InputElement
       elementDescriptor={descriptors.formFieldInput}
       elementId={descriptors.formFieldInput.setId(id)}
-      hasError={!!errorMessage}
+      hasError={debouncedState.feedbackType === 'error'}
       isDisabled={isDisabled}
       isRequired={isRequired}
       {...inputElementProps}
+      onFocus={e => {
+        inputElementProps.onFocus?.(e);
+        setTimeout(() => {
+          setIsFocused(true);
+        }, 350);
```

Comment:
I'm refactoring this. It's about waiting a bit before reducing the height in case another error appears (since we debounce validation as well). It'll be clearer after the refactor.